src/java/picard/analysis/CollectAlignmentSummaryMetrics.java

```diff
@@ -116,6 +116,10 @@ public static void main(final String[] argv) {
                     "in the file are aligned then alignment summary metrics collection will fail.");
         }
 
+        if (super.REFERENCE_SEQUENCE != null && this.REFERENCE_SEQUENCE == null) {
```

Comment:
I'm surprised that one needs to do this. the logic in CommanlineProgram was supposed to take care of this exact issue. @jacarey can you comment?